src/fhirtypes/ElementDefinition.ts

```diff
@@ -602,6 +623,301 @@ export class ElementDefinition {
     };
   }
 
+  /**
+   * Fixes a value to an ElementDefinition
+   * @param {FixedValueType} value - The value to fix
+   * @throws {NoSingleTypeError} when the ElementDefinition does not have a single type
+   * @throws {ValueAlreadyFixedError} when the value is already fixed to a different value
+   * @throws {MismatchedTypeError} when the value does not match the type of the ElementDefinition
+   */
+  fixValue(value: FixedValueType): void {
+    if (typeof value === 'boolean') {
+      this.fixBoolean(value);
+    } else if (typeof value === 'number') {
+      this.fixNumber(value);
+    } else if (typeof value === 'string') {
+      this.fixString(value);
+    } else if (value instanceof FshCode) {
+      this.fixFshCode(value);
+    } else if (value instanceof FshQuantity) {
+      this.fixFshQuantity(value);
+    } else if (value instanceof FshRatio) {
+      this.fixFshRatio(value);
+    }
+  }
+
+  /**
+   * Checks if a primitive value can be fixed
+   * @param {boolean | string | number} value - The value to fix
+   * @param {boolean | string | number} currentElementValue - The current value of the element
+   * @param {string} elementType - The type of the element as a string
+   * @throws {ValueAlreadyFixedError} when the currentElementValue exists and is different than the new value
+   * @returns {boolean} true if no value exists or the new value matches the old
+   */
+  private checkIfFixable(
+    value: boolean | string | number,
+    currentElementValue: boolean | string | number,
+    elementType: string
+  ): boolean {
+    if (currentElementValue != null && currentElementValue !== value) {
+      throw new ValueAlreadyFixedError(value, elementType, currentElementValue);
+    }
+    return true;
```

Comment:
Since this can only return one value (`true`) and throws in all cases, we should consider making it a `void` function.  By stating that it returns a `boolean`, someone calling the function might be led to believe it returns `true`/`false` rather than `true/throw` -- and might structure their code around the misconception.

Reply:
I still feel like it's better as a `void` function, but I'm willing to be wrong if the rest of the team disagrees with me.  The new approach (`true|undefined`) does address my main concern.

Reply:
Ah sorry I meant to respond to this explaining my choice there. The issue with making the function `void` is that we then can't use the `checkIfFixable` function in an `if` statement, because we can't test a function that returns `void` for true or false. So I could make it `void`, but then I would have to move the `checkIfFixable` function out of the condition and into the `if` block itself. Which you could argue is a more sensible way of doing things, but I personally liked having all the checking logic in the condition of the `if`.